### ISSUE TITLE
Terminal window line wrapping affects custom problemMatchers

### ISSUE THREAD

**Issue by Lavabomb** [microsoft/vscode#72828]
Issue Type: <b>Bug</b>

I use a custom problemMatcher for ARM compiler support. When I bulid using a task within visual studio, I can get duplicated output by resizing my window to a smaller dimension. One duplicate is always a truncated form of another. Resizing my window to a larger dimension correctly parses problems (no duplicates). The problem occurs with both my build and test tasks.

Here is a snippet of my task configuration:

```
{
    "version": "2.0.0",
    "problemMatcher": [
        // ADS 1.2 and DS-5 compiler errors and warnings, regex modified from gpm_utl.pm, checked on regex101.com
        {
            "owner": "ads",
            "fileLocation": "relative",
            "pattern": {
                "regexp": "^\"(.*)\", line (\\d+): (Strict )?(Warning|Error): ([C|L]\\d+[E|U|W]|\\s+#\\d+):\\s+(.*)$",
                "file": 1,
                "line": 2,
                "severity": 4,
                "code": 5,
                "message": 6
            }
        },

        // GCC works fine as-is
       "$gcc",

        // MSVC output is relative
        {
            "base": "$msCompile",
            "fileLocation": "relative",
        }
    ],
      "tasks": [
        ... /* Several custom build and test tasks */
    ]
}
```

VS Code version: Code - Insiders 1.34.0-insider (63c1ff218ecf3c2b3df8df1eab8e963d3c1f1a50, 2019-04-24T07:09:37.888Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz (8 x 3591)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.94GB (8.23GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-perl|hen|1.17.0
python|ms-|2019.4.11987
cpptools|ms-|0.22.1

(1 theme extensions excluded)

</details>

**Comment by alexr00** [comment]
@Lavabomb can provide a gif or screen shots of what you're seeing please?

**Comment by Lavabomb** [comment]
![parsing_error](https://user-images.githubusercontent.com/49964089/56751513-1edc5b80-673b-11e9-926f-4d7b4e717e87.gif)
The steps I took to make this GIF are as follows:
1. Build with the file explorer closed.
2. Touch the file generating the warning.
3. Close and reopen VS code to clear the problems.
4. Rebuild with the file explorer open.

**Comment by genne** [comment]
I'm experiencing the same issue but with the `$msCompile` problem matcher:

Terminal output:
![image](https://user-images.githubusercontent.com/2142735/65884120-8e184480-e398-11e9-9851-37a206df6240.png)

Problems:
![image](https://user-images.githubusercontent.com/2142735/65884159-9f615100-e398-11e9-8d65-49f6a1b3286d.png)

**Comment by alexr00** [comment]
I added a heuristic for improving this with https://github.com/microsoft/vscode/issues/77475
This is a duplicate of that bug and should be fixed starting with 1.38.0.
